### PR TITLE
feat(rune/bench): add comprehensive benchmark suite

### PR DESCRIPTION
**Pull request by RidwanAdebosin**
## Why it matters

Performance of Rune’s backward pass is currently blocking some users. This PR adds systematic benchmarks to provide numbers for both forward and backward passes, enabling investigation and future optimization.

## What’s in this PR

- Adds a benchmark suite in `rune/bench/` for:
  - Forward and backward (grad) passes on key ops
  - JIT compilation benchmarks
  - PyTorch and JAX comparison scripts

## How to run

```bash
dune exec rune/bench/bench_rune.exe
uv run rune/bench/bench_pytorch.py
uv run rune/bench/bench_jax.py
```

This PR implements the full benchmark suite as described in the task, covering all required operations, backward/JIT passes, and comparative scripts for PyTorch and JAX.
For more details, see the [benchmark suite README](https://github.com/RidwanAdebosin/raven/blob/ridwan-adebosin-outreachy-fix-build/rune/bench/README.md).

### CODE REVIEW

File: opam
Comment (gabyfle) [inline opam]: I think that this file is not really related to neither this PR nor Raven

File: results.rune.md
Comment (gabyfle) [inline results.rune.md]: This file is empty

**Comment by tmattio** [comment]
@RidwanAdebosin I've merge the rune benchmarks in https://github.com/raven-ml/raven/commit/94cfac0a7981abae79392f9f5b091d881d5da651, putting you as a co-author as it builds on your work in this PR.

I've decided to only include gradient benchmarks for now, since JIT is not ready yet, and the forward-pass benchmarks are already handled by Nx. I'm also only including comparative benchmarks against PyTorch, there's no good reason to track performance against Jax for now, we can do this when we measure JIT performance when it lands.

Thank you for your work on this!

**Comment by RidwanAdebosin** [comment]
@tmattio  Thanks a lot! I’m glad to see the benchmarks merged and appreciate being added as a co-author. Makes perfect sense to focus on gradients for now. I’ll be happy to help once the JIT and JAX benchmarking work begins.
